spec/ndx-pose.extensions.yaml

```diff
@@ -92,27 +121,88 @@ groups:
       dtype: text
       doc: Version string of the software tool used.
       required: false
-  - name: nodes
+  groups:
+  - neurodata_type_inc: PoseEstimationSeries
+    doc: Estimated position data for each body part.
+    quantity: '*'
+  links:
+  - target_type: Skeleton
+    doc: Layout of body part locations and connections.
+- neurodata_type_def: TrainingFrame
+  neurodata_type_inc: NWBDataInterface
+  default_name: TrainingFrame
+  doc: Group that holds ground-truth position data for all instances of a skeleton in a single frame.
+  attributes:
+  - name: annotator
     dtype: text
-    dims:
-    - num_body_parts
-    shape:
-    - null
-    doc: Array of body part names corresponding to the names of the PoseEstimationSeries
-      objects within this group.
+    doc: Name of annotator who labeled the TrainingFrame.
+    required: false
+  groups:
+  - neurodata_type_inc: Instance
+    doc: Position data for a single instance of a skeleton in a single training frame.
+    quantity: '*'
+  - name: source_video
+    neurodata_type_inc: ImageSeries
+    doc: Path to original video file and frame used.
     quantity: '?'
-  - name: edges
+    attributes:
+    - name: frame_index
+      dtype: uint8
+      doc: Frame index of TrainingFrame in original video file.
+      required: true
+  - name: source_frame
+    neurodata_type_inc: Images
+    doc: Collection of images used for training (stored internally or externally).
+    quantity: 1
+    attributes:
+    - name: images_index
+      dtype: uint8
+      doc: Index of TrainingFrame in collection of frames stored in Images.
+      required: true
```

Comment:
Thanks for this change @roomrys and sorry for my delay in response. 

I think this needs a little more adjustment - by including the `ImageSeries` type and the `Images` type here, each `TrainingFrame` will contain a complete `ImageSeries` or `Images` object (with all images) nested within it. This will use a lot of space unnecessarily. I think a better option would be to *link* to a single `ImageSeries` or `Image` object stored elsewhere. Sorry that this kinda contradicts my previous suggestion!

```suggestion
  attributes:
  - name: source_video_frame_index
    dtype: uint8
    doc: Frame index of training frame in the original video `source_video`. If provided, then `source_video` is required.
    required: false
  links:
  - name: source_video
    target_type: ImageSeries
    doc: Link to an ImageSeries representing a video of training frames (stored internally or externally). Required if `source_video_frame_index` is provided.
    quantity: '?'
  - name: source_frame
    target_type: Image
    doc: Link to an internally stored image representing the training frame. The target Image should be stored in an Images type in the file.
    quantity: '?'
```

Ideally we would have `source_video_frame_index` be an attribute on the `source_video` link but that is not allowed, so I put `source_video_frame_index` as an attribute on the `TrainingFrame` group instead.

**Use case 1: store training frames as internal images**
Solution 1a:
- Store the training frame images as `Image` objects in an `Images` object (`Images` is a collection of `Image` objects which can be ordered or unordered)
- For each image, create a `TrainingFrame` object that links to the `Image` and contains other things.

Solution 1b:
- Store the training frame images in an `ImageSeries` object, which requires timestamps for each frame (they are thus stored ordered). This type is preferred if the training frames come from a video. The data of an `ImageSeries` object can be internally stored frames or file paths to one or more external video files.
- For each image, create a `TrainingFrame` object that links to the `ImageSeries`, contains a `source_video_frame_index` attribute that indicates which frame of the `ImageSeries` is used, and contains other things.

**Use case 2: store training frames as external images**
This is not easily supported in NWB. See https://github.com/NeurodataWithoutBorders/nwb-schema/issues/529 for a feature request.

**Use case 3: store training frames as an external video file**
See solution 1b above.

Does this cover all the use cases?

Reply:
Hi @roomrys , happy new year! Do you have any thoughts on the above comment? I would like to move forward on adding this support into the ndx-pose extension. Thanks.

Reply:
I am happy to implement and test the changes above if you agree.

Reply:
Hi @rly, Happy New Year! The above use cases seem to cover the basis. Of course, trial by fire might point out some things that we've missed. Green light: feel free take over from this point.